azurelinuxagent/ga/env.py

```diff
@@ -118,6 +118,8 @@ def __init__(self, osutil, protocol):
         self._osutil = osutil
         self._protocol = protocol
         self._try_remove_legacy_firewall_rule = False
+        self._log_initial_firewall_list = True
+        self._wait = self._osutil.get_firewall_will_wait()
```

Comment:
This function has the potential of raising, and if it raises, the whole environment thread will not be started. Could you add more error handling here to ensure that never happens? Thanks!

Reply:
look like It's not good place to initialize here. If some reason first time failed, we are not retrying every run. Additionally, if this raises, we shouldn't enable the firewall and updating the global flag `_enable_firewall` to false. Any Idea how we can achieve this in env thread.

Reply:
you can keep the original implementation, like initialize the wait inside the Operation function maybe?

Reply:
we may not be able to update the global flag `_enable_firewall ` from `default.py `file if it's inside Operation function, can we?

Reply:
I think it should be, but just to keep things simple, I dont mind keeping it as is. I honestly don't like the design of using a global flag and then to stop trying to set firewalls if we fail even once, technically we should be retrying every hour or so until we can get it in just to make sure the firewalls are up, but I guess that's a bigger change than your bug so I'm good with as is.

Reply:
Now I don't see a need for wait call to be outside of osutil functions, I reverted to original implementation